webtop/__init__.py

```diff
@@ -9,13 +9,41 @@
 import math
 import os
 import signal
+import socket
 import time
 from threading import Event
-from typing import Dict, Collection, Optional, Deque
+from typing import Dict, Collection, Optional, Deque, List, Any, Callable
 import yaml
 from yarl import URL
 
 
+class CustomResolver(aiohttp.resolver.AbstractResolver):
+    async def close(self) -> None:
+        await self.async_resolver.close()
+
+    def __init__(self, *args, custom_mappings: Optional[Dict[str, str]] = None, **kwargs):
+        super().__init__(*args, **kwargs)  # type: ignore
+        if custom_mappings is None:
+            self.custom_mappings: Dict[str, str] = {}
+        else:
+            self.custom_mappings = custom_mappings
+        self.async_resolver = aiohttp.resolver.AsyncResolver()  # type: ignore
+
+    async def resolve(self, host: str, port: int = 0, family: int = socket.AF_INET) -> List[Dict[str, Any]]:
```

Comment:
Port 0 doesn't exist- does it have a special meaning here?

Reply:
I'm just copying the function signature from [`aiohttp.resolver.AsyncResolver`](https://github.com/aio-libs/aiohttp/blob/3.5/aiohttp/resolver.py#L61-L74) since we keep an instance of that in our custom resolver. Not sure why they do that. Maybe 0 means that the caller should use the port in the URL, and anything else is a port override. I can dig through the source and find where `resolve` is being called if you'd like.